nncf/openvino/graph/model_transformer.py

```diff
@@ -195,16 +190,22 @@ def _insert_outputs(model: ov.Model, outputs: List[Tuple[ov.Output, int, Callabl
         :param outputs: list of tuples with ov.Output & port_id.
         :return: Model with new outputs.
         """
+        outputs_type = ov.Type.f32
         results = model.get_results()
         params = model.get_parameters()
 
         extra_model_outputs = []
         for output, port_id in outputs:
-            output_name = output.get_node().get_friendly_name()
-            # TODO: (KodiaqQ) check out the models with the Split
+            node_output = output
+            output_name = node_output.get_node().get_friendly_name()
             result_name = get_result_node_name(output_name, port_id)
-            result = opset.result(output, name=result_name)
-            OVModelTransformer._update_tensor_name([result.get_output_tensor(0)], result_name)
+
+            if node_output.get_element_type() != outputs_type:
+                node_output = opset.convert(output, destination_type=outputs_type)
```

Comment:
This affects outputs for the `then-else` bodies of the `if` model. The point is that we should collect all statistics using `fp32` precision. But some outputs may have a `boolean` format, not a numeric.
What should we do in this case, @kshpv, @alexsu52?

Reply:
I would suggest to add flexibility. I mean the insert output transformation command should has `dtype` parameter to specify return type. By default the return type is FP32.

Reply:
Done.